File: C/Problem0007.c
/**
 * Problem0007 - 10001st prime
 * 
 * https://github.com/agileshaw/Project-Euler
 **/

#include <stdio.h>
#include <stdbool.h>
#include <math.h>

bool checkPrime(int num)
{
    int i;

    for (i = 2; i <= sqrt(num); i++) {
        if (num % i == 0)
            return false;
    }

    return true;
}

int main()
{
    int num = 1, count = 0;

    while (count < 10001) {
        num++;
        if (checkPrime(num))
            count++;
    }
    
    printf("The 10001st prime is: %d\n", num);

    return 0;

}

File: Python/Problem0007.py
# Problem0007 - 10001st prime
#
# https://github.com/agileshaw/Project-Euler

import math;
import itertools;

def checkPrime(num):
    for i in itertools.islice(itertools.count(2), int(math.sqrt(num) - 1)):
        if num % i == 0:
            return False
    return True

if __name__ == "__main__":
    count = 0
    for num in itertools.islice(itertools.count(2), None):
        if checkPrime(num):
            count += 1
        if count == 10001:
            break
    print("The 10001st prime is: %d" % num)
        
